internal/cssc/cssc.go

```diff
@@ -108,11 +134,66 @@ func GetFilterFromFilePath(filePath string) (Filter, error) {
 	return filter, nil
 }
 
+// Validates the filter and returns an error if the filter is invalid
+func (filter *Filter) ValidateFilter() error {
+	fmt.Println("Validating filter...")
+	if filter.Version == "" || filter.Version != "v1" {
+		return errors.New("Version is required in the filter and should be v1")
+	}
+	if filter.Repositories == nil || len(filter.Repositories) == 0 {
+		return errors.New("Repositories is required in the filter")
+	}
+	if filter.TagConvention != "" {
+		err := filter.TagConvention.IsValid()
+		if err != nil {
+			return err
+		}
+	}
+
+	allErrors := []string{}
+	for _, repo := range filter.Repositories {
+		if repo.Repository == "" {
+			return errors.New("Repository is required in the filter")
+		}
+		if repo.Tags == nil || len(repo.Tags) == 0 {
+			return errors.New("Tags is required in the filter")
+		}
+
+		incorrectTags := []string{}
+		for _, tag := range repo.Tags {
+			if tag == "" {
```

Comment:
Do we want to validate that the tag is even a valid tag filter? (There are regexes to check this, same for the repo)

Reply:
As discussed, we don't need it here.